pkg/controller/controller.go

```diff
@@ -247,6 +247,15 @@ func worker(queue workqueue.RateLimitingInterface, resourceType string, maxRetri
 	}
 }
 
+// operationError is a user-facing error that can be easily embedded in a
```

Comment:
This looks like a nice addition, but I wonder if it could be a followup or another PR from the refactor to keep the focus of the refactor.

Reply:
This piece greatly with the refactor, because it allows us to pass the condition-setting logic off to `processServiceInstanceOperationError` instead of handling it in every place this error's returned.